app/controllers/projects_controller.rb

```diff
@@ -206,6 +206,8 @@ def projectcommit
   #todo - /tree/master/file_name and generalize for /tree/branch/file_name
 
   def masterbranch
+    authorize! :update_image, @project
```

Comment:
That actually eliminates the access to the page itself not just the update button :)